### PR TITLE
Rspec v2

### PR DESCRIPTION
**Pull request by NataMarkova**
Hi @jrmcgarvey! This PR consists of non-completed rspec for orders and assignments. I've got an error for the "Orders get edit_order_path doesn't render the :edit template for a nil role":

`Expected response to be a redirect to <http://www.example.com/users/sign_in> but was a redirect to <http://www.example.com/users/sign_in?locale=en>.
 Expected "http://www.example.com/users/sign_in" to be === "http://www.example.com/users/sign_in?locale=en".`

The expected result for this test is `"http://www.example.com/users/sign_in?locale=en".` including query "locale=en" but the actual path should be "http://www.example.com/en/users/sign_in" without query. Is this a bug in routs?

### CODE REVIEW

**Comment by NataMarkova** [comment]
This is not completed test suites. After fixing the issue with the nil role, I'm going to add more tests.

**Comment by NataMarkova** [comment]
Another issue I have is the test "Assignments get show_assignment_path redirects to the customer index path if the assignment id is invalid". The test throws the error:
`Expected response to be a redirect to <http://www.example.com/en/customers> but was a redirect to <http://www.example.com/en/customers/5000>.
  Expected "http://www.example.com/en/customers" to be === "http://www.example.com/en/customers/5000".`
Where is my mistake here?

**Comment by jrmcgarvey** [comment]
Natalia, if you are not logged in, the assignments controller will redirect you to the sign in path.  Oh, also, when you create a PR and it is not complete, you should create it in draft mode so that it is not merged early.  Also, for the nil role test for edit order, it appears that the user is not signed in, so you are not really testing nil role.

File: app/helpers/application_helper.rb [hunk 1]
```diff
@@ -1,12 +1,23 @@
...
+      else
+        flash[:error] = "You are not authorize for this operation."
+        redirect_to services_path
+      end
+  end
+
+  def is_user_authorized_order?
+    if intern? || admin?
+      return 
+    elsif volunteer?
```
Comment (jrmcgarvey) [inline app/helpers/application_helper.rb:15]: Volunteers should get no access to the orders controller, so this elsif should be removed

File: app/helpers/application_helper.rb [hunk 2]
```diff
@@ -1,12 +1,23 @@
...
+      @customers = current_user.customers
+    else
+      flash[:error] = "You are not authorize for this operation."
+      redirect_to new_user_session_path
```
Comment (jrmcgarvey) [inline app/helpers/application_helper.rb:19]: The user is already signed in, so it does not make sense to redirect to the new_user_session_path.  In this case, you should redirect to the home_index_path.

File: app/controllers/orders_controller.rb [hunk 3]
```diff
@@ -76,4 +78,34 @@ def catch_not_found(e)
         redirect_to customers_path
       end
     end
+
+    def admin?
```
Comment (jrmcgarvey) [inline app/controllers/orders_controller.rb:82]: It seems like you are creating these methods over and over again for each controller.  You should put them in ApplicationController.rb, so that all controllers could inherit a single set of methods.  This is the DRY principal: Don't Repeat Yourself.

File: app/controllers/orders_controller.rb [hunk 4]
```diff
@@ -1,7 +1,9 @@
 class OrdersController < ApplicationController
+  include ApplicationHelper
   rescue_from ActiveRecord::RecordNotFound, with: :catch_not_found
   before_action :set_order, only: [:show, :edit, :update, :destroy]
   before_action :authenticate_user!
+  before_action :is_user_authorized_order?
```
Comment (jrmcgarvey) [inline app/controllers/orders_controller.rb:6]: You do not want the ? here.  That is for a method that returns true or false.

File: app/helpers/application_helper.rb [hunk 5]
```diff
@@ -1,12 +1,23 @@
...
-          return 
-        else
-          flash[:error] = "You are not authorize for this operation."
-          redirect_to services_path
-        end
+  def is_user_authorized?
+      if intern? || admin?
+        return 
+      else
+        flash[:error] = "You are not authorize for this operation."
```
Comment (jrmcgarvey) [inline app/helpers/application_helper.rb:7]: Misspelling here.  Should be "You are not authorized for this operation."

**Review by jrmcgarvey — CHANGES_REQUESTED** [review]
Some changes needed, see comments

File: app/controllers/orders_controller.rb (hunk 3 above)
Comment (NataMarkova) [reply app/controllers/orders_controller.rb:82]: The DRY principle is applied

File: app/controllers/orders_controller.rb (hunk 4 above)
Comment (NataMarkova) [reply app/controllers/orders_controller.rb:6]: Done

File: app/helpers/application_helper.rb (hunk 5 above)
Comment (NataMarkova) [reply app/helpers/application_helper.rb:7]: Misspelling is corrected.

File: app/helpers/application_helper.rb (hunk 1 above)
Comment (NataMarkova) [reply app/helpers/application_helper.rb:15]: Done.

File: app/helpers/application_helper.rb (hunk 2 above)
Comment (NataMarkova) [reply app/helpers/application_helper.rb:19]: Done.

**Review by jrmcgarvey — APPROVED** [review]
Ok
